.prettierrc.json

```diff
@@ -0,0 +1,16 @@
+{
+  "tabWidth": 2,
+  "printWidth": 120,
+  "trailingComma": "all",
+  "singleQuote": true,
+  "semi": false,
+  "overrides": [
+    {
+      "files": "contracts/**/*.sol",
+      "options": {
+        "tabWidth": 4,
+        "singleQuote": false
+      }
+    }
+  ]
+}
```

Comment:
does this auto adjust the code or is this more a check?

Reply:
The default part (for ts) is included in the eslint as an extension, so we will run auto lint in `lint-stage` with `eslint --fix`

The overrides part for .sol files will also be run in `lint-stage` with `prettier --write`

```json
# package.json
"lint-staged": {
    "**/*.sol": [
      "prettier --write",
      "git add"
    ],
    "test/**": [
      "eslint --fix",
      "git add"
    ]
  },
```